lib/puppet/indirector/file_bucket_file/file.rb

```diff
@@ -115,7 +115,11 @@ def validate_key(request)
     # @param files_original_path [String]
     #
     def matches(paths_file, files_original_path)
-      Puppet::FileSystem.open(paths_file, 0640, 'a+') do |f|
+      # Puppet will have already written the paths_file in the systems encoding
+      # given its possible that request.options[:bucket_path] or Puppet[:bucketdir]
+      # contained characters in an encoding that are not represented the
+      # same way when the bytes are decoded as UTF-8, continue using system encoding
+      Puppet::FileSystem.open(paths_file, 0640, "a+:#{Encoding.default_external.name}") do |f|
```

Comment:
Also I can't imagine `Encoding.default_external` changing again in ruby any time soon but if we were ever to adopt such a ruby we may want to remember to warn users about possible side effects in areas like this (file bucket).